ci-operator/step-registry/cucushift/installer/check/vsphere/zone/cucushift-installer-check-vsphere-zone-commands.sh

```diff
@@ -0,0 +1,159 @@
+#!/bin/bash
+
+set -o nounset
+#set -o errexit
+set -o pipefail
+set -x
+export KUBECONFIG=${SHARED_DIR}/kubeconfig
+source "${SHARED_DIR}/govc.sh"
+source "${SHARED_DIR}/vsphere_context.sh"
+INSTALL_CONFIG="${SHARED_DIR}/install-config.yaml"
+
+readarray -t zones_name_from_config < <(yq-go r ${INSTALL_CONFIG} 'platform.vsphere.failureDomains[*].name')
+readarray -t zones_region_from_config < <(yq-go r ${INSTALL_CONFIG} 'platform.vsphere.failureDomains[*].region')
+readarray -t zones_zone_from_config < <(yq-go r ${INSTALL_CONFIG} 'platform.vsphere.failureDomains[*].zone')
+readarray -t zones_datacenter_from_config < <(yq-go r ${INSTALL_CONFIG} 'platform.vsphere.failureDomains[*].topology.datacenter')
+
+function check_node_distribution() {
+    check_result=0
+    local node_type=$1
+    if [[ ${node_type} == "master" ]];then
+        readarray -t  node_distribution < <(yq-go r ${INSTALL_CONFIG} 'controlPlane.platform.vsphere.zones[*]')
+    else
+        readarray -t  node_distribution < <(yq-go r ${INSTALL_CONFIG} 'compute[*].platform.vsphere.zones[*]')
+    fi
+    mapfile -t  node_lists < <(oc get node --no-headers | grep ${node_type} | awk '{print $1}')
+    for node in "${node_lists[@]}"; do
+        # shellcheck disable=SC2207
+        lable_info=($(oc get node ${node} --show-labels --no-headers | awk -F, '{print $(NF-1),$NF}'))
+        node_region=$(echo ${lable_info[0]} | awk -F= '{print $2}')
+        node_zone=$(echo ${lable_info[1]} | awk -F= '{print $2}')
+        for ((i=0;i<${#zones_zone_from_config[@]};i++)); do
```

Comment:
why not directly use the element in `zones_zone_from_config` array? e.g:
```
for node_zone_from_config in ${zones_zone_from_config[@]}; do
    ....
done
```

Reply:
previously I check master and worker together. so if zone and region was match. also need to judge if zone-name belong to controlPlane.platform.vsphere.vsphere[*] or compute[*].platform.vsphere.zones[*].
so zones_name_from_config[${i}] is also needed.  after add the node_type parameter, seems no need to judge this check now:
if [[ ${node_distribution[*]} =~ ${zones_name_from_config[${i}]} ]];
I will update.